textdocument.vala

```diff
@@ -0,0 +1,40 @@
+class Vls.TextDocument : Object {
+    public weak Compilation compilation { get; private set; }
+    private string filename;
+
+    public Vala.SourceFile file { get; private set; }
+    public string uri { get; private set; }
+    public int version;
```

Comment:
What is the `version` used for?

Reply:
The language client sends the document version to us, and we can use this to check against our current version. It's useful for rejecting out-of-date updates, and it could be useful for caching later on.